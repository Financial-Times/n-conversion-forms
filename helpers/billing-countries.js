import { billingCountries } from 'n-common-static-data';

export { billingCountries };
