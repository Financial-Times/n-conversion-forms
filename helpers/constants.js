const printRegions = {
	cemeaV1: 'CEMEA_V1',
	cemeaV2: 'CEMEA_V2',
	apac: 'APAC',
};

const addressLine1AutoComplete = 'address-line1';
const addressLine2AutoComplete = 'address-line2';
const addressLine3AutoComplete = 'address-line3';

module.exports = {
	printRegions,
	addressLine1AutoComplete,
	addressLine2AutoComplete,
	addressLine3AutoComplete,
};
