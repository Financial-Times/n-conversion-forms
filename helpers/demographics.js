const { demographics } = require('n-common-static-data');

export { demographics };
