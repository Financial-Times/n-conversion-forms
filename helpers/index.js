module.exports = {
	'ncf-countries': require('./ncf-countries'),
	'ncf-common-data': require('./ncf-common-data'),
	cemeaV1ISO: require('./supportedCountries').cemeaV1ISO,
	cemeaV2ISO: require('./supportedCountries').cemeaV2ISO,
	apacISO: require('./supportedCountries').apacISO,
	identifyFTShippingZone: require('./supportedCountries')
		.identifyFTShippingZone,
	countriesSupported: require('./supportedCountries').countriesSupported,
	countriesSupportedISO: require('./supportedCountries').countriesSupportedISO,
	supportedPostcodeValidators: require('./supportedPostcodeValidators')
		.supportedPostcodeValidators,
	allSupportedPostcodeExamples: require('./supportedPostcodeExamples')
		.allSupportedPostcodeExamples,
	demographics: require('./demographics').demographics,
	billingCountries: require('./billing-countries').billingCountries,
};
