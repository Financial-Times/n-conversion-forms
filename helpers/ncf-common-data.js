const _get = require('lodash.get');
const COMMON_DATA = require('n-common-static-data');

module.exports = function ({ hash, fn }) {
	const context = Object.assign({}, { [hash.export]: _get(COMMON_DATA, hash.import) }, this);
	return fn(context);
};
