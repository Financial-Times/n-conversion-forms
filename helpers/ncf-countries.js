const { countries } = require('n-common-static-data').billingCountries;

const MINIMUM_TO_GROUP = 50;
const MINIMUM_TO_SHOW_GROUPS = 2;

module.exports = function ({ hash = {}, fn }) {
	const filter = Array.isArray(hash.filterList) ? hash.filterList : false;
	const value = hash.value;

	let data = countries;

	// Only show countries in the filter
	if (filter) {
		data = data.filter(item => filter.includes(item.code));
	}

	// Select the country
	if (value) {
		data = data.map(item => {
			item.selected = item.code === value;
			return item;
		});
	}

	// Group countries
	if (data.length >= MINIMUM_TO_GROUP) {
		data = groupCountries(data);
	}

	const context = Object.assign({ countries: data }, this);
	return fn(context);
};

/**
 * Produce a frequently used option group
 * @param {Array} countries
 * @return {Array}
 */
function groupCountries (countries) {
	const frequentlyUsed = ['GBR', 'USA', 'JPN', 'FRA', 'CAN'];

	const frequentlyUsedCountries = countries
		.filter(item => frequentlyUsed.includes(item.code))
		.map(item => Object.assign({}, item))
		.sort((a, b) => frequentlyUsed.indexOf(a.code) - frequentlyUsed.indexOf(b.code));

	// Not worth showing return standard countries list
	if (frequentlyUsedCountries.length < MINIMUM_TO_SHOW_GROUPS) {
		return countries;
	}

	// If selected country is in frequently used only leave that one selected
	// otherwise the user is shown the lowest selected option
	const selected = frequentlyUsedCountries.find(item => item.selected);
	if (selected) {
		countries.forEach(item => item.selected = false);
	}

	return [
		{ label: 'Frequently Used', countries: frequentlyUsedCountries },
		{ label: 'Alphabetical', countries }
	];
}
