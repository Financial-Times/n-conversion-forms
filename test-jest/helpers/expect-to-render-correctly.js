import ReactDOMServer from 'react-dom/server';

export const expectToRenderCorrectly = {
	toRenderCorrectly(reactComponent, props) {
		// Render the React component with props.
		const ReactComponentRendered = ReactDOMServer.renderToStaticMarkup(
			reactComponent(props)
		);

		// eslint-disable-next-line jest/no-standalone-expect
		expect(ReactComponentRendered).toMatchSnapshot();

		// This point is reached when the above assertions were successful.
		// The test should therefore always pass, that means it needs to be
		// `true` when used normally, and `false` when `.not` was used.
		return {
			pass: !this.isNot,
		};
	},
};
