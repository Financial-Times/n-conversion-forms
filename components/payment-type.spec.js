import { PaymentType } from './index';
import { expectToRenderCorrectly } from '../test-jest/helpers/expect-to-render-correctly';

expect.extend(expectToRenderCorrectly);

describe('PaymentType', () => {
	it('render with default props', () => {
		const props = {};

		expect(PaymentType).toRenderCorrectly(props);
	});

	it('render with enableApplepay', () => {
		const props = {
			enableApplepay: true,
		};

		expect(PaymentType).toRenderCorrectly(props);
	});

	it('render with enableCreditcard', () => {
		const props = {
			enableCreditcard: true,
		};

		expect(PaymentType).toRenderCorrectly(props);
	});

	it('render with enableDirectdebit', () => {
		const props = {
			enableDirectdebit: true,
		};

		expect(PaymentType).toRenderCorrectly(props);
	});

	it('render with enablePaypal', () => {
		const props = {
			enablePaypal: true,
		};

		expect(PaymentType).toRenderCorrectly(props);
	});

	it('render with value', () => {
		const props = {
			value: 'paypal',
		};

		expect(PaymentType).toRenderCorrectly(props);
	});

	it('can initialise with the loader visible', () => {
		const props = {
			showLoaderOnInit: true,
		};

		expect(PaymentType).toRenderCorrectly({}, props);
	});

	it('render with isSingleTerm', () => {
		const props = {
			isSingleTerm: true,
		};

		expect(PaymentType).toRenderCorrectly(props);
	});

	it('render with isSingleTermChecked', () => {
		const props = {
			isSingleTerm: true,
			isSingleTermChecked: true,
		};

		expect(PaymentType).toRenderCorrectly(props);
	});
});
