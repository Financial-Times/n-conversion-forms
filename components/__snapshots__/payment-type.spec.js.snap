// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentType render with default props 1`] = `
<div class="ncf__security-seal">
</div>
<div id="paymentTypeField"
     class="o-forms-field ncf__field"
>
  <div class="o-forms-input o-forms-input--radio-box ncf__payment-type-selector">
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--creditcard ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="creditcard"
               id="creditcard"
               aria-label="Credit / Debit Card"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Credit / Debit Card
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--paypal ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="paypal"
               id="paypal"
               aria-label="PayPal"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          PayPal
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--directdebit ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="directdebit"
               id="directdebit"
               aria-label="Direct Debit"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Direct Debit
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--applepay ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="applepay"
               id="applepay"
               aria-label="Apple Pay"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Apple Pay
        </span>
      </label>
    </div>
  </div>
  <div class="o-forms-input__error">
    Please enter a valid payment type
  </div>
</div>
`;

exports[`PaymentType render with default props 2`] = `
<div class="ncf__security-seal">
</div>
<div id="paymentTypeField"
     class="o-forms-field ncf__field"
>
  <div class="o-forms-input o-forms-input--radio-box ncf__payment-type-selector">
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--creditcard ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="creditcard"
               id="creditcard"
               aria-label="Credit / Debit Card"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Credit / Debit Card
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--paypal ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="paypal"
               id="paypal"
               aria-label="PayPal"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          PayPal
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--directdebit ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="directdebit"
               id="directdebit"
               aria-label="Direct Debit"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Direct Debit
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--applepay ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="applepay"
               id="applepay"
               aria-label="Apple Pay"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Apple Pay
        </span>
      </label>
    </div>
  </div>
  <div class="o-forms-input__error">
    Please enter a valid payment type
  </div>
</div>
`;

exports[`PaymentType render with enableApplepay 1`] = `
<div class="ncf__security-seal">
</div>
<div id="paymentTypeField"
     class="o-forms-field ncf__field"
>
  <div class="o-forms-input o-forms-input--radio-box ncf__payment-type-selector">
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--creditcard ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="creditcard"
               id="creditcard"
               aria-label="Credit / Debit Card"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Credit / Debit Card
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--paypal ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="paypal"
               id="paypal"
               aria-label="PayPal"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          PayPal
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--directdebit ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="directdebit"
               id="directdebit"
               aria-label="Direct Debit"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Direct Debit
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--applepay">
      <label>
        <input type="radio"
               name="paymentType"
               value="applepay"
               id="applepay"
               aria-label="Apple Pay"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Apple Pay
        </span>
      </label>
    </div>
  </div>
  <div class="o-forms-input__error">
    Please enter a valid payment type
  </div>
</div>
`;

exports[`PaymentType render with enableApplepay 2`] = `
<div class="ncf__security-seal">
</div>
<div id="paymentTypeField"
     class="o-forms-field ncf__field"
>
  <div class="o-forms-input o-forms-input--radio-box ncf__payment-type-selector">
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--creditcard ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="creditcard"
               id="creditcard"
               aria-label="Credit / Debit Card"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Credit / Debit Card
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--paypal ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="paypal"
               id="paypal"
               aria-label="PayPal"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          PayPal
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--directdebit ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="directdebit"
               id="directdebit"
               aria-label="Direct Debit"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Direct Debit
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--applepay">
      <label>
        <input type="radio"
               name="paymentType"
               value="applepay"
               id="applepay"
               aria-label="Apple Pay"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Apple Pay
        </span>
      </label>
    </div>
  </div>
  <div class="o-forms-input__error">
    Please enter a valid payment type
  </div>
</div>
`;

exports[`PaymentType render with enableCreditcard 1`] = `
<div class="ncf__security-seal">
</div>
<div id="paymentTypeField"
     class="o-forms-field ncf__field"
>
  <div class="o-forms-input o-forms-input--radio-box ncf__payment-type-selector">
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--creditcard">
      <label>
        <input type="radio"
               name="paymentType"
               value="creditcard"
               id="creditcard"
               aria-label="Credit / Debit Card"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Credit / Debit Card
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--paypal ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="paypal"
               id="paypal"
               aria-label="PayPal"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          PayPal
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--directdebit ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="directdebit"
               id="directdebit"
               aria-label="Direct Debit"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Direct Debit
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--applepay ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="applepay"
               id="applepay"
               aria-label="Apple Pay"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Apple Pay
        </span>
      </label>
    </div>
  </div>
  <div class="o-forms-input__error">
    Please enter a valid payment type
  </div>
  <div class="ncf__payment-type-panel ncf__payment-type-panel--creditcard ncf__payment-type-panel--directdebit n-ui-hide">
    <div class="ncf__zuora-payment-overlay n-ui-hide">
    </div>
    <div id="zuora_payment"
         class="ncf__zuora-payment"
    >
      <iframe id="z_hppm_iframe"
              title="Zuora Payment"
              overflow="visible"
              scrolling="no"
              frameborder="0"
              allowtransparency="true"
              class="z_hppm_iframe"
              style="display:block"
      >
      </iframe>
    </div>
    <script src="https://static.zuora.com/Resources/libs/hosted/1.3.1/zuora-min.js">
    </script>
  </div>
</div>
`;

exports[`PaymentType render with enableCreditcard 2`] = `
<div class="ncf__security-seal">
</div>
<div id="paymentTypeField"
     class="o-forms-field ncf__field"
>
  <div class="o-forms-input o-forms-input--radio-box ncf__payment-type-selector">
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--creditcard">
      <label>
        <input type="radio"
               name="paymentType"
               value="creditcard"
               id="creditcard"
               aria-label="Credit / Debit Card"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Credit / Debit Card
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--paypal ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="paypal"
               id="paypal"
               aria-label="PayPal"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          PayPal
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--directdebit ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="directdebit"
               id="directdebit"
               aria-label="Direct Debit"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Direct Debit
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--applepay ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="applepay"
               id="applepay"
               aria-label="Apple Pay"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Apple Pay
        </span>
      </label>
    </div>
  </div>
  <div class="o-forms-input__error">
    Please enter a valid payment type
  </div>
  <div class="ncf__payment-type-panel ncf__payment-type-panel--creditcard ncf__payment-type-panel--directdebit n-ui-hide">
    <div class="ncf__zuora-payment-overlay n-ui-hide">
    </div>
    <div id="zuora_payment"
         class="ncf__zuora-payment"
    >
      <iframe id="z_hppm_iframe"
              title="Zuora Payment"
              overflow="visible"
              scrolling="no"
              frameborder="0"
              allowtransparency="true"
              class="z_hppm_iframe"
              style="display:block"
      >
      </iframe>
    </div>
    <script src="https://static.zuora.com/Resources/libs/hosted/1.3.1/zuora-min.js">
    </script>
  </div>
</div>
`;

exports[`PaymentType render with enableDirectdebit 1`] = `
<div class="ncf__security-seal">
</div>
<div id="paymentTypeField"
     class="o-forms-field ncf__field"
>
  <div class="o-forms-input o-forms-input--radio-box ncf__payment-type-selector">
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--creditcard ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="creditcard"
               id="creditcard"
               aria-label="Credit / Debit Card"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Credit / Debit Card
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--paypal ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="paypal"
               id="paypal"
               aria-label="PayPal"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          PayPal
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--directdebit">
      <label>
        <input type="radio"
               name="paymentType"
               value="directdebit"
               id="directdebit"
               aria-label="Direct Debit"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Direct Debit
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--applepay ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="applepay"
               id="applepay"
               aria-label="Apple Pay"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Apple Pay
        </span>
      </label>
    </div>
  </div>
  <div class="o-forms-input__error">
    Please enter a valid payment type
  </div>
  <div class="ncf__payment-type-panel ncf__payment-type-panel--directdebit n-ui-hide">
    <div id="directDebitGuarantee"
         class="ncf__directdebit-guarantee"
         data-o-component="o-expander"
         data-o-expander-shrink-to="hidden"
         data-o-expander-expanded-toggle-text="guarantee"
         data-o-expander-collapsed-toggle-text="guarantee"
         role="tabpanel"
    >
      <p>
        Direct Debit is only supported in the UK
      </p>
      <p>
        Your payments are protected by the Direct Debit
        <button type="button"
                class="ncf__directdebit-guarantee-toggle"
        >
          guarantee
        </button>
      </p>
      <ul class="ncf__directdebit-guarantee-list">
        <li>
          This Guarantee is offered by all banks and building societies that accept instructions to pay Direct Debits.
        </li>
        <li>
          If there are any changes to the amount, date or frequency of your Direct Debit GoCardless re: The Financial Times will notify you 3 working days in advance of your account being debited or as otherwise agreed. If you request GoCardless Ltd re: The Financial Times to collect a payment, confirmation of the amount and date will be given to you at the time of the request.
        </li>
        <li>
          If an error is made in the payment of your Direct Debit, by GoCardless Ltd re: The Financial Times or your bank or building society, you are entitled to a full and immediate refund of the amount paid from your bank or building society.
        </li>
        <li>
          If you receive a refund you are not entitled to, you must pay it back when GoCardless Ltd re: The Financial Times asks you to.
        </li>
        <li>
          You can cancel a Direct Debit at any time by simply contacting your bank or building society. Written confirmation may be required. Please also notify us.
        </li>
      </ul>
    </div>
  </div>
  <div class="ncf__payment-type-panel ncf__payment-type-panel--creditcard ncf__payment-type-panel--directdebit n-ui-hide">
    <div class="ncf__zuora-payment-overlay n-ui-hide">
    </div>
    <div id="zuora_payment"
         class="ncf__zuora-payment"
    >
      <iframe id="z_hppm_iframe"
              title="Zuora Payment"
              overflow="visible"
              scrolling="no"
              frameborder="0"
              allowtransparency="true"
              class="z_hppm_iframe"
              style="display:block"
      >
      </iframe>
    </div>
    <script src="https://static.zuora.com/Resources/libs/hosted/1.3.1/zuora-min.js">
    </script>
  </div>
</div>
`;

exports[`PaymentType render with enableDirectdebit 2`] = `
<div class="ncf__security-seal">
</div>
<div id="paymentTypeField"
     class="o-forms-field ncf__field"
>
  <div class="o-forms-input o-forms-input--radio-box ncf__payment-type-selector">
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--creditcard ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="creditcard"
               id="creditcard"
               aria-label="Credit / Debit Card"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Credit / Debit Card
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--paypal ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="paypal"
               id="paypal"
               aria-label="PayPal"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          PayPal
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--directdebit">
      <label>
        <input type="radio"
               name="paymentType"
               value="directdebit"
               id="directdebit"
               aria-label="Direct Debit"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Direct Debit
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--applepay ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="applepay"
               id="applepay"
               aria-label="Apple Pay"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Apple Pay
        </span>
      </label>
    </div>
  </div>
  <div class="o-forms-input__error">
    Please enter a valid payment type
  </div>
  <div class="ncf__payment-type-panel ncf__payment-type-panel--directdebit n-ui-hide">
    <div id="directDebitGuarantee"
         class="ncf__directdebit-guarantee"
         data-o-component="o-expander"
         data-o-expander-shrink-to="hidden"
         data-o-expander-expanded-toggle-text="guarantee"
         data-o-expander-collapsed-toggle-text="guarantee"
         role="tabpanel"
    >
      <p>
        Direct Debit is only supported in the UK
      </p>
      <p>
        Your payments are protected by the Direct Debit
        <button type="button"
                class="ncf__directdebit-guarantee-toggle"
        >
          guarantee
        </button>
      </p>
      <ul class="ncf__directdebit-guarantee-list">
        <li>
          This Guarantee is offered by all banks and building societies that accept instructions to pay Direct Debits.
        </li>
        <li>
          If there are any changes to the amount, date or frequency of your Direct Debit GoCardless re: The Financial Times will notify you 3 working days in advance of your account being debited or as otherwise agreed. If you request GoCardless Ltd re: The Financial Times to collect a payment, confirmation of the amount and date will be given to you at the time of the request.
        </li>
        <li>
          If an error is made in the payment of your Direct Debit, by GoCardless Ltd re: The Financial Times or your bank or building society, you are entitled to a full and immediate refund of the amount paid from your bank or building society.
        </li>
        <li>
          If you receive a refund you are not entitled to, you must pay it back when GoCardless Ltd re: The Financial Times asks you to.
        </li>
        <li>
          You can cancel a Direct Debit at any time by simply contacting your bank or building society. Written confirmation may be required. Please also notify us.
        </li>
      </ul>
    </div>
  </div>
  <div class="ncf__payment-type-panel ncf__payment-type-panel--creditcard ncf__payment-type-panel--directdebit n-ui-hide">
    <div class="ncf__zuora-payment-overlay n-ui-hide">
    </div>
    <div id="zuora_payment"
         class="ncf__zuora-payment"
    >
      <iframe id="z_hppm_iframe"
              title="Zuora Payment"
              overflow="visible"
              scrolling="no"
              frameborder="0"
              allowtransparency="true"
              class="z_hppm_iframe"
              style="display:block"
      >
      </iframe>
    </div>
    <script src="https://static.zuora.com/Resources/libs/hosted/1.3.1/zuora-min.js">
    </script>
  </div>
</div>
`;

exports[`PaymentType render with enablePaypal 1`] = `
<div class="ncf__security-seal">
</div>
<div id="paymentTypeField"
     class="o-forms-field ncf__field"
>
  <div class="o-forms-input o-forms-input--radio-box ncf__payment-type-selector">
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--creditcard ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="creditcard"
               id="creditcard"
               aria-label="Credit / Debit Card"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Credit / Debit Card
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--paypal">
      <label>
        <input type="radio"
               name="paymentType"
               value="paypal"
               id="paypal"
               aria-label="PayPal"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          PayPal
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--directdebit ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="directdebit"
               id="directdebit"
               aria-label="Direct Debit"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Direct Debit
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--applepay ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="applepay"
               id="applepay"
               aria-label="Apple Pay"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Apple Pay
        </span>
      </label>
    </div>
  </div>
  <div class="o-forms-input__error">
    Please enter a valid payment type
  </div>
</div>
`;

exports[`PaymentType render with enablePaypal 2`] = `
<div class="ncf__security-seal">
</div>
<div id="paymentTypeField"
     class="o-forms-field ncf__field"
>
  <div class="o-forms-input o-forms-input--radio-box ncf__payment-type-selector">
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--creditcard ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="creditcard"
               id="creditcard"
               aria-label="Credit / Debit Card"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Credit / Debit Card
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--paypal">
      <label>
        <input type="radio"
               name="paymentType"
               value="paypal"
               id="paypal"
               aria-label="PayPal"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          PayPal
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--directdebit ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="directdebit"
               id="directdebit"
               aria-label="Direct Debit"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Direct Debit
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--applepay ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="applepay"
               id="applepay"
               aria-label="Apple Pay"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Apple Pay
        </span>
      </label>
    </div>
  </div>
  <div class="o-forms-input__error">
    Please enter a valid payment type
  </div>
</div>
`;

exports[`PaymentType render with value 1`] = `
<div class="ncf__security-seal">
</div>
<div id="paymentTypeField"
     class="o-forms-field ncf__field"
>
  <div class="o-forms-input o-forms-input--radio-box ncf__payment-type-selector">
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--creditcard ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="creditcard"
               id="creditcard"
               aria-label="Credit / Debit Card"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Credit / Debit Card
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--paypal ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="paypal"
               id="paypal"
               aria-label="PayPal"
               checked
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          PayPal
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--directdebit ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="directdebit"
               id="directdebit"
               aria-label="Direct Debit"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Direct Debit
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--applepay ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="applepay"
               id="applepay"
               aria-label="Apple Pay"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Apple Pay
        </span>
      </label>
    </div>
  </div>
  <div class="o-forms-input__error">
    Please enter a valid payment type
  </div>
</div>
`;

exports[`PaymentType render with value 2`] = `
<div class="ncf__security-seal">
</div>
<div id="paymentTypeField"
     class="o-forms-field ncf__field"
>
  <div class="o-forms-input o-forms-input--radio-box ncf__payment-type-selector">
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--creditcard ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="creditcard"
               id="creditcard"
               aria-label="Credit / Debit Card"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Credit / Debit Card
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--paypal ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="paypal"
               id="paypal"
               aria-label="PayPal"
               checked
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          PayPal
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--directdebit ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="directdebit"
               id="directdebit"
               aria-label="Direct Debit"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Direct Debit
        </span>
      </label>
    </div>
    <div class="o-forms-input--radio-box__container ncf__payment-type ncf__payment-type--applepay ncf__hidden">
      <label>
        <input type="radio"
               name="paymentType"
               value="applepay"
               id="applepay"
               aria-label="Apple Pay"
        >
        <span class="o-forms-input__label"
              aria-hidden="true"
        >
          Apple Pay
        </span>
      </label>
    </div>
  </div>
  <div class="o-forms-input__error">
    Please enter a valid payment type
  </div>
</div>
`;
